### PR TITLE
Add 7.12 branch

### PR DESCRIPTION
**Pull request by jrodewig**
Adds a 7.12 branch to the docs.

CI is expected to fail until all repos have a 7.12 branch.

### CODE REVIEW

File: shared/versions/stack/7.12.asciidoc [hunk 1]
```diff
@@ -0,0 +1,50 @@
+:version:                7.12.0
+////
+bare_version never includes -alpha or -beta
+////
+:bare_version:           7.12.0
+:logstash_version:       7.12.0
+:elasticsearch_version:  7.12.0
+:kibana_version:         7.12.0
+:apm_server_version:     7.12.0
+:branch:                 7.x
```
Comment (gtback) [inline shared/versions/stack/7.12.asciidoc:10]: This should be 7.12 now, right?

**Review by gtback — CHANGES_REQUESTED** [review]
Looks like this might be missing adding the new version to a couple books that were added in #2031 (just merged last week).

- Enterprise Search Python client (line ~983)
- Enterprise Search Ruby client (line ~998

File: shared/versions/stack/7.12.asciidoc (hunk 1 above)
Comment (jrodewig) [reply shared/versions/stack/7.12.asciidoc]: Yes. Fixed with 597151adcbde64bd7433a839653d940fe742b7f6

**Comment by jrodewig** [comment]
Thanks @gtback. I addressed your feedback in 597151adcbde64bd7433a839653d940fe742b7f6.

**Review by gtback — APPROVED** [review]

**Comment by jrodewig** [comment]
@elasticmachine update branch

**Comment by jrodewig** [comment]
@elasticmachine run elasticsearch-ci/docs

**Pull request merged**